genjava/client/okhttp_impl.go

```diff
@@ -132,13 +127,13 @@ func (g *Generator) generateClientMethod(w *sources.Writer, operation *spec.Name
 		if !response.Type.Definition.IsEmpty() {
 			responseJavaType := g.Types.Java(&response.Type.Definition)
 			w.Line(`%s responseBody;`, responseJavaType)
-			responseBody := g.Models.ReadJson("response.body().string()", responseJavaType)
+			responseBody, exception := g.Models.ReadJson("response.body().string()", responseJavaType)
 			if response.Type.Definition.Plain == spec.TypeString {
 				responseBody = `response.body().string()`
 			}
 			generateClientTryCatch(w,
 				fmt.Sprintf(`responseBody = %s;`, responseBody),
-				`IOException`, `e`,
+				exception, `e`,
```

Comment:
Separate string body and json body. String body should still use `IOException`. While JSON body read should catch exception that depends on JSON libs.